### DISCUSSION TITLE
Run "pnpm dev &" seems not working

### DISCUSSION THREAD

**Discussion by RBVoidLin** [kwchang0831/svelte-QWER#52] [Q&A] [answered]
Hi, I recently find this amazing blog template.
I followed the teaching article and deployed on oracle vps and everything is OK.
![image](https://user-images.githubusercontent.com/113545499/217276029-ff06750d-86ee-4371-b9c8-abd41e664c33.png)
However, when I tried to run "pnpm dev" as background service, something seems weird.
After I run "pnpm dev &" , console looks the same as previous image.
![image](https://user-images.githubusercontent.com/113545499/217278104-5f0398e5-5368-4fb0-a282-ffebeed9b530.png)
As result, I can not access the website any more.

Anyone know what could be happening?

**Comment by kwchang0831** [comment, marked as answer]
My env:
- System:
  - OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)

## My first attempt was trying to reproduce your situation. 

`pnpm dev &`
```
  VITE v4.1.4  ready in 2041 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: (Omitted)
```

Right after, check the job state.

`jobs -l`
```
Job     Group   CPU     State   Command
1       32655   2%      stopped pnpm dev &
```

I cannot get it to work as background process this way.

## My second attempt was to try [pm2](https://pm2.keymetrics.io/).

This works, I can run `pnpm dev` as background process.
This is how I did it.

Install pm2 using npm globally.
```
npm i pm2 -g
```

Run `pnpm dev` with pm2. (be sure to be in the right directory)
```
pm2 start "pnpm dev" --name QWER
```

Check jobs details, Ctrl+C to leave
```
pm2 monit
```

See realtime logs for all jobs
```
pm2 logs
```

Delete job, you can see Job id in `pm2 ls` or `pm2 monit`
```
pm2 delete [JobID]
```

**Comment by RBVoidLin** [comment]
Thanks for your response !!
I also provide a tricky solution here

1. I write a shellscript : It will check whether the pid of pnpm alive ,cd blog folder and run "pnpm dev"
2. Put it into cron job and check it periodically.  
```
*/2 * * * * /root/start_up_rblog.sh
```

And it works 🙃
